File: Configuration/src/main/java/com/medecineWebApp/Configuration/models/setting/Calendar.java
package com.medecineWebApp.Configuration.models.setting;

import com.medecineWebApp.Configuration.models.Auditable;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.io.Serializable;
import java.util.List;
@Entity
@Table(name = "calendar")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EntityListeners(AuditingEntityListener.class)
public class Calendar extends Auditable implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name; // Example: "Doctor's Calendar"

    @OneToMany(mappedBy = "calendar", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Event> events;

    // Add a user association if required (e.g., a Doctor or Patient)
    private Long userId;
}

File: Client/frontend/src/app/models/calendar.ts
import { IEvent } from "./event";

export interface ICalendar{
    id?:number;
    name?:string;
    events?:IEvent[];
    userId?:number;
}

export class Calendar implements ICalendar{
    constructor(
        public id?:number,
        public name?:string,
        public events?:IEvent[],
        public userId?:number
    ){}
}    